src/librustc/session/config.rs

```diff
@@ -109,6 +109,7 @@ impl OutputType {
             OutputType::Bitcode => "bc",
             OutputType::Assembly => "s",
             OutputType::LlvmAssembly => "ll",
+            OutputType::Object if cfg!(target_os = "windows") => "obj",
             OutputType::Object => "o",
```

Comment:
@eddyb Oh, it could be part of `TargetOptions`. Is there a reason that `Object` is part of `OutputType`, and `TargetOptions` (which has `exe_suffix`, `staticlib_suffix`...) does not simply have `obj_suffix`?